pkg/core/naming/contextual_test.go

```diff
@@ -0,0 +1,90 @@
+package naming_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/core/naming"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+)
+
+type fakeResource struct{}
+
+func (f *fakeResource) GetMeta() core_model.ResourceMeta          { return nil }
+func (f *fakeResource) SetMeta(core_model.ResourceMeta)           {}
+func (f *fakeResource) GetSpec() core_model.ResourceSpec          { return nil }
+func (f *fakeResource) SetSpec(core_model.ResourceSpec) error     { return nil }
+func (f *fakeResource) GetStatus() core_model.ResourceStatus      { return nil }
+func (f *fakeResource) SetStatus(core_model.ResourceStatus) error { return nil }
+func (f *fakeResource) Descriptor() core_model.ResourceTypeDescriptor {
+	return core_model.ResourceTypeDescriptor{Name: "UnknownType"}
+}
+
+var _ = Describe("ContextualInboundName", func() {
+	It("should return error when resource is nil", func() {
+		// when
+		name, err := naming.ContextualInboundName(nil, "http")
+
+		// then
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("resource is nil"))
```

Comment:
nitpick: I know there's `MatchError` as well, does it give us anything better than this?